script/path.js

```diff
@@ -0,0 +1,361 @@
+			Path.outfit = {};
+		}
+		
+		// Add the armour row
+		var armour = _("none");
+		if($SM.get('stores["s armour"]', true) > 0)
+			armour = _("steel");
+		else if($SM.get('stores["i armour"]', true) > 0)
+			armour = _("iron");
+		else if($SM.get('stores["l armour"]', true) > 0)
+			armour = _("leather");
+		var aRow = $('#armourRow');
+		if(aRow.length === 0) {
+			aRow = $('<div>').attr('id', 'armourRow').addClass('outfitRow').prependTo(outfit);
+			$('<div>').addClass('row_key').text(_('armour')).appendTo(aRow);
+			$('<div>').addClass('row_val').text(armour).appendTo(aRow);
+			$('<div>').addClass('clear').appendTo(aRow);
+		} else {
+			$('.row_val', aRow).text(armour);
+		}
+		
+		// Add the water row
+		var wRow = $('#waterRow');
+		if(wRow.length === 0) {
+			wRow = $('<div>').attr('id', 'waterRow').addClass('outfitRow').insertAfter(aRow);
+			$('<div>').addClass('row_key').text(_('water')).appendTo(wRow);
+			$('<div>').addClass('row_val').text(World.getMaxWater()).appendTo(wRow);
+			$('<div>').addClass('clear').appendTo(wRow);
+		} else {
+			$('.row_val', wRow).text(World.getMaxWater());
+		}
+		
+		var space = Path.getFreeSpace();
+		var currentBagCapacity = 0;
+		// Add the non-craftables to the craftables
+		var carryable = $.extend({
+			'cured meat': { type: 'tool', desc: _('restores') + ' ' + World.MEAT_HEAL + ' ' + _('hp') },
+			'bullets': { type: 'tool', desc: _('use with rifle') },
+			'grenade': {type: 'weapon' },
+			'bolas': {type: 'weapon' },
+			'laser rifle': {type: 'weapon' },
+			'energy cell': {type: 'tool', desc: _('use with laser rifle') },
+			'bayonet': {type: 'weapon' },
+			'charm': {type: 'tool'},
+			'medicine': {type: 'tool', desc: _('restores') + ' ' + World.MEDS_HEAL + ' ' + _('hp') }
+		}, Room.Craftables);
+		
+		for(var k in carryable) {
+			var lk = _(k);
+			var store = carryable[k];
+			var have = $SM.get('stores["'+k+'"]');
+			var num = Path.outfit[k];
+			num = typeof num == 'number' ? num : 0;
+			if (have !== undefined) {
+				if (have < num) { num = have; }
+				$SM.set(k, num, true);
+			}
+
+			var row = $('div#outfit_row_' + k.replace(' ', '-'), outfit);
+			if((store.type == 'tool' || store.type == 'weapon') && have > 0) {
+				currentBagCapacity += num * Path.getWeight(k);
+				if(row.length === 0) {
+					row = Path.createOutfittingRow(k, num, store, store.name);
+					
+					var curPrev = null;
+					outfit.children().each(function(i) {
+						var child = $(this);
+						if(child.attr('id').indexOf('outfit_row_') === 0) {
+							var cName = child.children('.row_key').text();
+							if(cName < lk) {
+								curPrev = child.attr('id');
+							}
+						}
+					});
+					if(curPrev == null) {
+						row.insertAfter(wRow);
+					} else {
+						row.insertAfter(outfit.find('#' + curPrev));
+					}
+				} else {
+					$('div#' + row.attr('id') + ' > div.row_val > span', outfit).text(num);
+					$('div#' + row.attr('id') + ' .tooltip .numAvailable', outfit).text(have - num);
```

Comment:
Some of these css rules could be change preventers if some of the front end layout is changed. css classes and ids would be better to use than selectors for html elements like "div" and "span".